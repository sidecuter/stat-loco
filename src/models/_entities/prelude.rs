//! `SeaORM` Entity, @generated by sea-orm-codegen 1.1.10

pub use super::sites::Entity as Sites;
pub use super::user_ids::Entity as UserIds;
pub use super::users::Entity as Users;
