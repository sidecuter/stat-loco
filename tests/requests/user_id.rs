use insta::assert_debug_snapshot;
use loco_rs::testing::prelude::*;
use serial_test::serial;
use stat_api::app::App;

// use crate::requests::prepare_data;

macro_rules! configure_insta {
    ($($expr:expr),*) => {
        let mut settings = insta::Settings::clone_current();
        settings.set_prepend_module_to_snapshot(false);
        settings.set_snapshot_suffix("user_id_request");
        let _guard = settings.bind_to_scope();
    };
}

#[tokio::test]
#[serial]
async fn can_get_user_id() {
    configure_insta!();
    request::<App, _, _>(|request, _ctx| async move {
        let res = request.get("/api/user_ids/new").await;
        assert_eq!(res.status_code(), 200, "Creation request should succeed");
        insta::with_settings!({
            filters => vec![
                (
                    r"([0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12})",
                    "PID",
                ),
                (
                    r"\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2} \+\d{2}:\d{2}",
                    "DATE",
                ),
            ]
        }, {
            assert_debug_snapshot!(res.text());
        });
    })
    .await;
}

// #[tokio::test]
// #[serial]
// async fn can_get_user_ids() {
//     configure_insta!();
//     request::<App, _, _>(|request, ctx| async move {
//         let user = prepare_data::init_user_login(&request, &ctx).await;
// TODO: create prepare_data for user_id requests
//         let (auth_key, auth_value) = prepare_data::auth_header(&user.token);
//         let response = request
//             .get("/api/user_ids/1")
//             .add_header(auth_key, auth_value)
//             .await;
//         assert_eq!(
//             response.status_code(),
//             200,
//             "Authorization should be succesfull"
//         );
//         // insta::with_settings!({
//         //     filters => vec![
//         //         (
//         //             r"([0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12})",
//         //             "PID",
//         //         ),
//         //         (
//         //             r"\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2} \+\d{2}:\d{2}",
//         //             "DATE",
//         //         ),
//         //     ]
//         // }, {
//         //     assert_debug_snapshot!(res.text());
//         // });
//     })
//     .await;
// }
