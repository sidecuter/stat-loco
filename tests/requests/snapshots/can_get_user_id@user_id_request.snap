---
source: tests/requests/user_id.rs
expression: res.text()
---
"{\"user_id\":\"PID\",\"created_at\":\"DATE\"}"
